MovieNight/MovieNightModel.swift

```diff
@@ -29,32 +30,65 @@ enum User {
     case Crab
 }
 
-protocol Person {
+protocol Nameable {
     var name: String { get }
-    var movies: [Movie]? { get }
 }
 
-struct Actor: Person, JSONDecodable {
+struct Actor: JSONDecodable, Equatable {
     var name: String
-    var movies: [Movie]?
+    var id: Int
+    
+    init(name: String, id: Int) {
+        self.name = name
+        self.id = id
+    }
+
     
     init?(JSON: [String: AnyObject]) {
         guard let name = JSON["name"] as? String else {
             return nil
         }
+        guard let id = JSON["id"] as? Int else {
+            return nil
+        }
         self.name = name
+        self.id = id
+    }
+    
+    static func ==(lhs: Actor, rhs: Actor) -> Bool {
+        return lhs.id == rhs.id
     }
 }
 
-struct Movie : JSONDecodable {
+struct Credit {
+    var movieId: Int
+    var actor: Actor
+    
+    init?(JSON: [String: AnyObject], movieId: Int) {
+        guard let name = JSON["name"] as? String else {
+            return nil
+        }
+        guard let id = JSON["id"] as? Int else {
+            return nil
+        }
+        
+        self.actor = Actor(name: name, id: id)
+        self.movieId = movieId
+    }
+}
+
+struct Movie : JSONDecodable, Hashable, Equatable {
     var title: String
     var releaseDate: String?
     var voteAverage: Int?
+    var genreIds: [Int]
     var id: Int
+    var hashValue : Int { return self.id }
```

Comment:
Hmm, gonna check it, but what I have in mind is that the hashValue is just an int associated to an instance, so I don't see why this would be wrong, but that's the first time I use it, so I'm gonna read about it. Tks